src/istio/control/http/client_context.cc

```diff
@@ -14,25 +14,57 @@
  */
 
 #include "src/istio/control/http/client_context.h"
+#include "include/istio/utils/attribute_names.h"
 
+using ::istio::mixer::v1::Attributes_AttributeValue;
 using ::istio::mixer::v1::config::client::ServiceConfig;
+using ::istio::utils::AttributeName;
+using ::istio::utils::CreateLocalAttributes;
 
 namespace istio {
 namespace control {
 namespace http {
+const char* kReporterOutbound = "outbound";
+
+namespace {
+
+// isOutbound returns true if this is an outbound listener configuration.
+// It relies on pilot setting context.reporter.kind == outbound;
+static bool isOutbound(
+    const ::istio::mixer::v1::config::client::HttpClientConfig& config) {
+  bool outbound = false;
+  const auto& attributes_map = config.mixer_attributes().attributes();
+  const auto it = attributes_map.find(AttributeName::kContextReporterKind);
+  if (it != attributes_map.end()) {
+    const Attributes_AttributeValue& value = it->second;
+    if (kReporterOutbound == value.string_value()) {
+      outbound = true;
+    }
+  }
+  return outbound;
+}
+
+}  // namespace
 
 ClientContext::ClientContext(const Controller::Options& data)
     : ClientContextBase(data.config.transport(), data.env),
       config_(data.config),
-      service_config_cache_size_(data.service_config_cache_size) {}
+      service_config_cache_size_(data.service_config_cache_size),
+      outbound_(isOutbound(data.config)) {
+  local_attributes_ = CreateLocalAttributes(data.local_node);
```

Comment:
this local_attributes_ can be just directly embeded object.  not a unique_ptr.

ClientContext {

   LocalAttributes local_attributes_;
}

at Constructor 


  CreateLocalAttributes(local_node, &local_attributes_);
}